evaluator/evaluate.go

```diff
@@ -207,21 +207,29 @@ func analyzePackages(tx *gorm.DB, system *models.SystemPlatform, vmaasData vmaas
 	return installed, updatable, nil
 }
 
-func analyzeRepos(tx *gorm.DB, system *models.SystemPlatform, vmaasData vmaas.UpdatesV2Response) (
+func analyzeRepos(tx *gorm.DB, system *models.SystemPlatform) (
 	thirdParty bool, err error) {
 	if !enableRepoAnalysis {
 		utils.Log().Debug("repo analysis disabled, skipping")
 		return false, nil
 	}
 
-	// if system has associated more repos then there are in vmaasData
-	// one of them has to be thirdparty
-	repoCount, err := countRepos(tx, system.RhAccountID, system.ID)
+	// if system has associated at least one third party repo
+	// it's marked as third party system
+	var thirdPartyCount int
+	err = tx.Table("system_repo sr").
+		Joins("join repo r on r.id = sr.repo_id").
+		Where("sr.rh_account_id = ?", system.RhAccountID).
+		Where("sr.system_id = ?", system.ID).
+		Where("r.third_party = true").
+		Count(&thirdPartyCount).Error
 	if err != nil {
-		return false, errors.Wrap(err, "Unable to load repo data")
+		utils.Log("err", err.Error(), "accountID", system.RhAccountID, "systemID", system.ID).
+			Warn("counting third party repos")
+		return false, err
 	}
-	thirdParty = len(vmaasData.RepositoryList) < repoCount
-	return thirdParty, nil
+
+	return (thirdPartyCount > 0), nil
```

Comment:
I would rather see it in a variable, but it's up to you.

Reply:
changed